vm/vmcore/guestmem/src/lib.rs

```diff
@@ -397,6 +397,14 @@ pub unsafe trait GuestMemoryAccess: 'static + Send + Sync {
         None
     }
 
+    /// Similar to access_bitmap, but uses the execute permission bitmap in
+    /// place of the read bitmap, e.g. for checking permissions for instruction
+    /// retrieval.
+    #[cfg(feature = "bitmap")]
+    fn execute_access_bitmap(&self, _access_type: ExecuteAccessType) -> Option<BitmapInfo> {
```

Comment:
I don't understand why `guestmem` has to change at all.

Reply:
What I had imagined is that within `underhill_mem` we'd just instantiate a `GuestMemoryAccess` implementation that uses the execute bitmap, rather than have `guestmem` know how to swap it in.

Reply:
I see. Yes, I think having a wrapper here is better than adding complexity to `guestmem` itself.